proto/ethermint/evm/v1/query.proto

```diff
@@ -81,6 +82,52 @@ service Query {
   rpc BaseFee(QueryBaseFeeRequest) returns (QueryBaseFeeResponse) {
     option (google.api.http).get = "/ethermint/evm/v1/base_fee";
   }
+
+  // -----------------------------------------
+  // Query clients for evm keeper statedb access
+  rpc QueryGetHashStateDB(GetHashRequest) returns (GetHashResponse) {
+    option (google.api.http).get = "/ethermint/evm/v1/get_hash";
+  }
+  
+  rpc PostAddBalanceStateDB(AddBalanceRequest) returns (AddBalanceResponse) {
+    option (google.api.http).post = "/ethermint/evm/v1/add_balance";
+  }
+  
+  rpc PostSubBalanceStateDB(SubBalanceRequest) returns (SubBalanceResponse) {
+    option (google.api.http).post = "/ethermint/evm/v1/sub_balance";
+  }
+
+  rpc QueryGetBalanceStateDB(GetBalanceRequest) returns (GetBalanceResponse) {
+    option (google.api.http).get = "/ethermint/evm/v1/get_balance";
+  }
+
+  rpc QueryGetAccountStateDB(GetAccountRequest) returns (GetAccountResponse) {
+    option (google.api.http).get = "/ethermint/evm/v1/get_account";
+  }
+
+  rpc QueryGetStateStateDB(GetStateRequest) returns (GetStateResponse) {
+    option (google.api.http).get = "/ethermint/evm/v1/get_state";
+  }
+
+  rpc QueryGetCodeStateDB(GetCodeRequest) returns (GetCodeResponse) {
+    option (google.api.http).get = "/ethermint/evm/v1/get_code";
+  }
+
+  rpc PostSetAccountStateDB(SetAccountRequest) returns (SetAccountResponse) {
```

Comment:
What happens if a end user queries this endpoint via gRPC, _during_ a tx transaction. Like some kind of race condition.

```
->--prepareTx-------------SGX execution--------SetAcountStateDB
                   ^
           user calls PostSetAccountStateDB
```

Won't the external user's `PostSetAccountStateDB` mess with the real SGX's `PostSetAccountStateDB`?

Reply:
@amaury1093 He will definitely mess the execution as it uses same context. We shouldn't give external user access to these endpoints.

Reply:
> We shouldn't give external user access to these endpoints.

How do we prevent that?

Reply:
@amaury1093 Seems like to be a challenging problem to solve. I think at least we should use different gRPC server to make it separated and should make it locally accessible only at least.